Source/Orts.Simulation/Simulation/RollingStocks/MSTSLocomotive.cs

```diff
@@ -1333,35 +1333,44 @@ public override void Initialize()
 
             // Check TrainBrakesControllerMaxSystemPressure parameter for "correct" value 
             // This is only done for vacuum brakes as the UoM can be confusing - it defaults to psi due to way parameter is read, and if units are entered then a InHG value can be incorrectly converted.
-            if ((BrakeSystem is VacuumSinglePipe) && TrainBrakeController.MaxPressurePSI > 10 && TrainBrakeController.MaxPressurePSI < 15)
+            if ((BrakeSystem is VacuumSinglePipe))
             {
-                Trace.TraceInformation("TrainBrakeController.MaxPressurePSI being incorrectly read as {0} Inhg, - set to value of {1} InHg", TrainBrakeController.MaxPressurePSI, Bar.ToInHg(Bar.FromPSI(TrainBrakeController.MaxPressurePSI)));
-                TrainBrakeController.MaxPressurePSI = Bar.ToInHg(Bar.FromPSI(TrainBrakeController.MaxPressurePSI));
+                if (TrainBrakeController.MaxPressurePSI == 21 || TrainBrakeController.MaxPressurePSI == 25) // If 21 or 25 has been entered assume that it is 21InHg or 25InHg, and convert it to the correct psi equivalent
+                {
+                    if (Simulator.Settings.VerboseConfigurationMessages)
+                    {
+                        Trace.TraceInformation("TrainBrakeController.MaxPressurePSI is assumed to be {0} Inhg, - confirmed as a value of {1} InHg", TrainBrakeController.MaxPressurePSI, Bar.ToInHg(Bar.FromPSI(Bar.ToPSI(Bar.FromInHg(TrainBrakeController.MaxPressurePSI)))));
```

Comment:
Hi Peter,

These conversions puzzle me. Why are we changing a value in InHg to bar and then to PSI and then to bar and finally to InHg? There must be something I'm missing.

Also, when units of measure are missing like this, how about adding a warning in the log that vacuum pressures are best defined with units (bar, psi, inHg) to avoid ambiguity.

Best wishes, 

Chris

Reply:
Hi Chris,

> These conversions puzzle me. Why are we changing a value in InHg to bar and then to PSI and then to bar and finally to InHg? There must be something I'm missing.

I agree it is a bit complicated.

I wanted to provide both the input value from the user, and the new default value created. So there are two ways to do this. The way that I have done it, and the other alternative is to temporarily save the user value, and this will reduce the number of conversion in one place.

In the interest of making the logic for the code clearer, I will change the commit to try and reflect the 2nd approach.
 
> Also, when units of measure are missing like this, how about adding a warning in the log that vacuum pressures are best defined with units (bar, psi, inHg) to avoid ambiguity.
The OR manual highlights the need to add UoM to parameters.

Indirectly the associated error messages are to highlight that there is a lack of clarity that OR has attempted to "fix". Most MSTS files would be configured without the UoM.